packages/eset_protect/data_stream/event/elasticsearch/ingest_pipeline/default.yml

```
@@ -454,6 +475,16 @@ processors:
       tag: rename_object_uri
       target_field: eset_protect.event.object_uri
       ignore_missing: true
+  - set:
+      field: file.path
+      if: "ctx.eset_protect?.event?.object_uri != null && ctx.eset_protect?.event?.object_type?.toLowerCase() == 'file'"
+      copy_from: eset_protect.event.object_uri
+  - grok: 
+      field: file.path
+      patterns: 
+        - '^(file\:\/\/\/)?(?<file.drive_letter>\w)\:(?<file.directory>\/.*\/)?(?<file.name>.+(\.(?<file.extension>.+))?)?$'
+        - '.*\/(?<file.name>.+\.(?<file.extension>.+))$'
```

```suggestion
        - '^(file:///)?(?<file.drive_letter>\w)\:(?<file.directory>/.*/)?(?<file.name>.+(\.(?<file.extension>.+))?)?$'
        - '.*/(?<file.name>.+\.(?<file.extension>.+))$'
```
Note that strictly speaking the file scheme has a potential for a host in the slot between `file://` and `/`. This probably does not matter here.

Given the comments above, maybe
```
  - grok: 
      field: file.path
      patterns: 
        - '^(file:///)?%{FILEPATH:file.path}$'
        - '.*/(?<file.name>.+\.(?<file.extension>.+))$'
      pattern_definitions:
        FILEPATH: '(?<file.drive_letter>\w):(?<file.directory>/.*/|/)?(?<file.name>.+(\.(?<file.extension>.+))?)?'
      if: ctx.file?.path != null
```

---

I will test this ^

---

I think the more dramatic change should be made.